src/gseproject/templates/game/scan.html

```diff
@@ -9,5 +9,201 @@
+        { name: "Devonshire House", location: { lat: 50.73494324268427, lng: -3.534412366357129 } },
+        { name: "Queens Building", location: { lat: 50.73449636375293, lng: -3.5346366654312993 } },
+        { name: "Old Library", location: { lat: 50.733423968372335, lng: -3.534261025796802 } },
+        { name: "Hatherly Laboratories", location: { lat: 50.73403019226748, lng: -3.5331800569656564 } },
+        { name: "Exeter Northcott Theatre", location: { lat: 50.73581601104062, lng: -3.535169181217625 } },
+        { name: "Peter Chalk Centre", location: { lat: 50.736291440802674, lng: -3.5359763384371847 } },
+        { name: "Laver Building", location: { lat: 50.737386322677416, lng: -3.5334687416050254 } },
+        { name: "Harrison Building", location: { lat: 50.737507526268054, lng: -3.532889295495887 } },
+        { name: "SWIOT Building", location: { lat: 50.737908686355524, lng: -3.5304422433978604 } },
+        { name: "Innovation Centre", location: { lat: 50.73826130515783, lng: -3.531193928475741 } },
+        { name: "XFi Building", location: { lat: 50.736058501012415, lng: -3.5302494398729363 } },
+        { name: "Building:One", location: { lat: 50.73571302267035, lng: -3.529901119084657 } },
+        { name: "Streatham Court", location: { lat: 50.7358055379937, lng: -3.5313437765806253 } },
+        { name: "Lemon Grove", location: { lat: 50.73503423639206, lng: -3.529732856785485 } }
+        // Add more buildings as needed
+    ];
+
+    // Function to calculate the distance between two points
+    function calculateDistance(lat1, lon1, lat2, lon2) {
+        const R = 6371000; // Radius of the Earth in meters
+        const dLat = (lat2 - lat1) * (Math.PI / 180); // Convert degrees to radians
+        const dLon = (lon2 - lon1) * (Math.PI / 180);
+        const a =
+            Math.sin(dLat / 2) * Math.sin(dLat / 2) +
+            Math.cos(lat1 * (Math.PI / 180)) * Math.cos(lat2 * (Math.PI / 180)) *
+            Math.sin(dLon / 2) * Math.sin(dLon / 2);
+        const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
+        const distance = R * c; // Distance in meters
+        return distance;
+    }
+
+    // Function to find buildings within the specified range
+    function findBuildingsWithinRange(latitude, longitude, maxRange) {
+        const userLocation = { lat: latitude, lng: longitude };
+        const buildingsWithinRange = [];
+
+        buildings.forEach((building) => {
+            const distance = calculateDistance(
+                userLocation.lat,
+                userLocation.lng,
+                building.location.lat,
+                building.location.lng
+            );
+
+            if (distance <= maxRange) {
+                buildingsWithinRange.push(building);
+            }
+        });
+
+        return buildingsWithinRange;
+    }
+
+    // Function to handle scanning and checking location
+    // Function to handle scanning and checking location
+    function scanForCoordinate() {
```

Comment:
If these functions are in here, they need to be removed from base_game_template.js

Reply:
This file is from marco. Please deny this and check his pull